### DISCUSSION TITLE
What is the problem on packing for StudentList?

### DISCUSSION THREAD

**Discussion by zyanbingchn** [protobuf-c/protobuf-c#638] [Q&A] [answered]
(1) student.proto
message Student
{
  required int32 id = 1;
  required string name = 2;         
  optional string desc = 3;         
};

message StudentList
{
  repeated  Student students = 1;
};

(2) student.pb-c.h
typedef struct _Student Student;
typedef struct _StudentList StudentList;

struct  _Student
{
  ProtobufCMessage base;
  int32_t id;
  char *name;
  char *desc;
};

struct  _StudentList
{
  ProtobufCMessage base;
  size_t n_students;
  Student **students;
};

(3) demo.c
Student** pTmpStudents = (Student**)malloc(sizeof(Student*) * 10);
int j = 0;
for (j = 0; j < 10; j++)
{
	    pTmpStudents[j] = malloc(sizeof(Student));
 	    student_list__init(pTmpStudents[j]);
		_pTmpStudents[j]->id = 123;_
}

StudentList studentListObj = STUDENT_LIST__INIT;
studentListObj.n_students = 10;
studentListObj.students = pTmpStudents;

(4) Segmentation fault when get packed size
unsigned int packedDataLen = student_list__get_packed_size(&studentListObj);

if delete code:
  pTmpStudents[j]->id = 123;
then it is ok!

**Comment by zyanbingchn** [comment, marked as answer]
I am sorry!
it is my low level code error, 
now close the issue.

**Comment by edanla** [comment]
How do you allocate space and assign values to strings，
I have the same problem，
Can you explain it in detail.Thank you.
